pep517/envbuild.py

```diff
@@ -14,13 +15,35 @@
 
 log = logging.getLogger(__name__)
 
+
 def _load_pyproject(source_dir):
     with open(os.path.join(source_dir, 'pyproject.toml')) as f:
         pyproject_data = pytoml.load(f)
     buildsys = pyproject_data['build-system']
     return buildsys['requires'], buildsys['build-backend']
 
 
+REQUIREMENT_PATTERN = re.compile(r'^(?P<name>[^~><=!]+)(?P<spec>.*)$')
+
```

Comment:
should we use instead here the pep 440 parser instead of the naive regex? I've used it when I hit this edge case in tox to resolve the conflict. Tbh though I think we should really warn if the two versions don't match exactly. it's likely a bug.

Reply:
@pfmoore @takluyver Thoughts?

Reply:
I suspect so, but I don't use this class in pip, so I've no real idea.

Reply:
AFAICT there are no existing PEP 440 implementations that can tell if a version specifier has conflicts by looking at it. So the only viable thing to do IMO is to just feed it into pip and see what happens. 

Theoratically the input here can be an arbitrary PEP 508-compliant string, complete with markers and such, so the complete solution should likely use [packaging.requirements] to do the parsing. I can do that if @takluyver thinks it’s a good idea to add `packaging` as a dependency.

Reply:
https://packaging.pypa.io/en/latest/requirements/ should allow you to parse, and then you can make the decision if the two versions (if both exists) matches or not